### PR TITLE
v1.13 Backports 2023-01-18

### PR DESCRIPTION
**Pull request by aanm**
* [ ] #22811 -- gh/workflows: ci-datapath updates (@brb)
 * [ ] #22985 -- datapath: Fix L7 ingress with XDP (@brb)
 * [x] #22894 -- k8s: Add node-role.kubernetes.io/control-plane taint (@sayboras)
 * [ ] #22884 -- Fix crash of CES queue delay metric when CESTracker is nil (@dlapcevic)
 * [x] #22622 -- feat(hubble): add L7 verdicts to hubble_policy_verdicts_total metric (@raphink)
 * [x] #22977 -- iptables: skip reverse IP lookup (@jibi)
 * [x] #22972 -- bpf: Fix LB loopback path with ingress policy (@aditighag)
 * [x] #22994 -- hubble: Fix panic if IP address cannot be parsed (@gandro)
 * [x] #23018 -- bpf: test: fix xdp_lb4_forward_to_other_node test (@julianwiedmann)
 * [ ] #22979 -- build: Avoid re-building when building docs from the main Makefile (@jrajahalme)
 * [x] #22325 -- gha: Bump k8s version in kind conformance tests (@sayboras)
 * [x] #21768 -- Endpoint/CEP Ownership Fix (@tommyp1ckles)
 * [x] #23007 -- gha: Improve coverage for Ingress/GatewayAPI (@sayboras)
 * [x] #23029 -- bpf: nodeport: reset EDT aggregate ID for XDP-to-TC tunnel punt (@julianwiedmann)
 * [x] #22950 -- per-node config improvements: testing, null selector, cleanups (@squeed)
 * [x] #23037 -- ctmap: fix-up host_local flag in the DSR NAT entry for GC test (@julianwiedmann)
 * [ ] #22962 -- clustermesh: Add missing brackets of etcd option (@YutaroHayakawa)
 * [x] #23043 -- images: Update Hubble CLI to v0.11.0 (@gandro)
 * [x] #22810 -- nodeport: fix CT accounting in revDNAT path (@julianwiedmann)
 * [ ] #22191 -- Generate Software Bill of Materials during release (@sandipanpanda)
 * [ ] #22379 -- docs: move star wars demo to getting started (@yoyo-go)
 * [ ] #23076 -- datapath: Get rid of NO_REDIRECT define (@brb)
 * [ ] #22768 -- Make cilium pprof listen address configurable (@chancez)
 * [x] #22919 -- bpf: Relax constant check for dst_id for clang 14+ (@sayboras)
 * [x] #23075 -- bpf: lb: catch write error in lb6_xlate() (@julianwiedmann)
 * [x] #23077 -- envoy: Fix lock leak in config validation failure (@joestringer)
 * [ ] #22795 -- pkg/metrics: include revision and arch info in cilium_version (@ArthurChiao)
 * [x] #23063 -- bpf: lb: fix L3 pseudo-hdr csum update for SCTP in __lb6_rev_nat() (@julianwiedmann)
 * [ ] #22825 -- gh/workflows: Extend ci-datapath config to include lb-mode and endpoint-routes (@brb)
 * [x] #23119 -- bpf: nodeport: fix-up error check in rev_nodeport_lb*() for XDP (@julianwiedmann)

PRs skipped due conflicts / regressions:

 * #22425 -- gh/workflows: tune LVH VM params (@brb)
 * #22928 -- CODEOWNERS: Cover test/bpf_tests by sig-datapath (@christarazi)
 * #23020 -- ci: extend nat46x64 l4lb test suite (@borkmann)
 * #21980 -- datapath: Fix L7 reply to outside when endpoint routes disabled (@brb)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 22811 22985 22894 22884 22622 22977 22972 22994 23018 22979 22325 21768 23007 23029 22950 23037 22962 23043 22810 22191 22379 23076 22768 22919 23075 23077 22795 23063 22825 23119; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
$ make add-label BRANCH=v1.13 ISSUES=22811,22985,22894,22884,22622,22977,22972,22994,23018,22979,22325,21768,23007,23029,22950,23037,22962,23043,22810,22191,22379,23076,22768,22919,23075,23077,22795,23063,22825,23119
```

### CODE REVIEW

**Comment by aanm** [comment]
/test-backport-1.13

**Review by joestringer — APPROVED** [review]
Backport of my change LGTM.

File: .github/workflows/tests-smoke.yaml
Comment (joestringer) [inline .github/workflows/tests-smoke.yaml]: This doesn't seem correct for a v1.13 backport?

**Review by joestringer — COMMENTED** [review]
I looked back over @cilium/github-sec since apparently I gave approval for that codeowner. Would be good to get @brb to look at the datapath workflow changes to make sure all the PRs are included for fixing those. Otherwise one minor nit below in a comment.

File: .github/workflows/tests-smoke.yaml
Comment (joestringer) [reply .github/workflows/tests-smoke.yaml]: (comment so it's not functional, maybe it's easier to just have this stay the same as master to minimize diff/conflicts in future)

Comment (aanm) [reply .github/workflows/tests-smoke.yaml]: actually this was likely caused by me. I'll revert it

**Comment by aanm** [comment]
/test-backport-1.13

**Review by sayboras — APPROVED** [review]
Thanks and looks good on my commits :100:

**Comment by aditighag** [comment]
Looks good for my commit. Thanks!

**Review by tommyp1ckles — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

**Review by raphink — APPROVED** [review]
LGTM

**Review by jibi — APPROVED** [review]

**Review by raphink — APPROVED** [review]

**Review by squeed — APPROVED** [review]

**Comment by aanm** [comment]
/test-backport-1.13

**Review by gandro — APPROVED** [review]
My commits looks good, thanks!

**Comment by aanm** [comment]
/test-1.25-4.19

**Review by YutaroHayakawa — APPROVED** [review]

**Comment by aanm** [comment]
/test-1.22-4.9

**Comment by ldelossa** [comment]
Failures:
 ConformanceKind1.19 / installation-and-connectivity (pull_request) Failing after 18m  - https://github.com/cilium/cilium/issues/22435
 
  Cilium L4LB XDP (ci-l4lb-1.13) — L4LB test failed - https://github.com/cilium/cilium/issues/22933

  ConformanceGKE (ci-gke-1.13) — Connectivity test failed - Seems unrelated, no existing issue tho.

```
    Run while [ "$(gcloud container operations list --filter="status=RUNNING AND targetLink~cilium-cilium-3948012883" --format="value(name)")" ];do
    ERROR: (gcloud.container.clusters.delete) Some requests did not succeed:
     - args: ["ResponseError: code=404, message=Not found: projects/***/zones/us-west2-a/clusters/cilium-cilium-3948012883.\nNo cluster named 'cilium-cilium-3948012883' in ***."]
       exit_code: 1
```

   k8s-1.20-kernel-4.9 (test-1.20-4.9) — Build finished. - https://github.com/cilium/cilium/issues/15455

   k8s-1.21-kernel-4.9 (test-1.21-4.9) — Build finished. - https://github.com/cilium/cilium/issues/22578

   k8s-1.24-kernel-4.9 (test-1.24-4.9) — Build finished. - https://github.com/cilium/cilium/issues/22578

   k8s-1.25-kernel-4.19 (test-1.25-4.19) — Build finished. - https://github.com/cilium/cilium/issues/22578

**Comment by joestringer** [comment]
@ldelossa were you able to triage all of the failures? When I see that many red x's, I get a bit concerned. I'm not specifically familiar with the v1.13 tree stability right now but that seems like more failures than should be there. Do we think that v1.13 is already unstable, or is there some potential that there is something in these backports that introduce some of these failures?

**Comment by joestringer** [comment]
I've updated https://isogo.to/ci on the second page to see v1.13 branch stability:

image

**Comment by ldelossa** [comment]
Those failed tests all had existing issues here: https://github.com/cilium/cilium/pull/23147#issuecomment-1396105062 no?

**Comment by joestringer** [comment]
Ah sorry I missed some of it, but I see five different tests failing across four jobs:

https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/1520/
https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/2287/
https://jenkins.cilium.io/job/Cilium-PR-K8s-1.24-kernel-4.9/12/
https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-4.19/465/

**Comment by joestringer** [comment]
I think I was partly just unfamiliar with the way you laid out the failures, but also when I look at the middle two failures above for instance, I can see failures outside of the hostfw tests.

**Comment by joestringer** [comment]
I guess that if not all jobs are failing the same way, that suggests that the tests are unreliable to some degree (otherwise they would consistently pass on all jobs or fail on all jobs), but the summary seems to suggest that the last three links have all the same failure.

**Comment by joestringer** [comment]
Ah now that I'm looking closer, there are a lot more jobs running than I thought. At first I saw this:

image

The above yells to me "don't merge, something is wrong". But scrolling down further, I see this, which is more reassuring:

image

**Comment by joestringer** [comment]
Given the failure rate, I think it's unlikely the above failures are representing a reliable failure in v1.13. But it would be good to at least also link/file the related issues for the Istio / Upgrade tests. This is the primary way that we can tell in future v1.13 backports whether the job was stable or not. If we don't keep track now, then next time we won't know if the issue is newly introduced by that subsequent backport. Would you mind following up for the [k8s-1.21](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/2287/) and [k8s-1.24](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.24-kernel-4.9/12/) jobs?

**Comment by ldelossa** [comment]
https://github.com/cilium/cilium/issues/23175
https://github.com/cilium/cilium/issues/23174

**Pull request merged**
